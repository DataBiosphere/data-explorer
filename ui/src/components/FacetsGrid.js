import "./FacetsGrid.css";
import FacetCard from "./FacetCard";

import React, { Component } from "react";
import { GridList, GridTile } from "material-ui/GridList";

function FacetsGrid(props) {
  const facets = props.facets;
  const updateFacets = props.updateFacets;
  const facetsList = facets.map(facet => (
    <GridTile key={facet.name}>
      <FacetCard facet={facet} updateFacets={updateFacets} />
    </GridTile>
  ));
  return (
    <GridList className="gridList" cols={3} cellHeight="auto" padding={1}>
      {facetsList}
    </GridList>
  );
}

export default FacetsGrid;
