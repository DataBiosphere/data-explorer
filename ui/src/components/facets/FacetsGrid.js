import React from "react";
import GridList from "@material-ui/core/GridList";
import GridListTile from "@material-ui/core/GridListTile";
import { withStyles } from "@material-ui/core/styles";

import HistogramFacet from "components/facets/HistogramFacet";
import TimeSeriesFacet from "components/facets/TimeSeriesFacet";

const styles = {
  root: {
    // Needed to make left box shadow show up for all left-most tiles
    overflow: "unset"
  },
  tile: {
    // Needed to make left box shadow show up for all tiles
    overflow: "unset"
  }
};

function FacetsGrid(props) {
  const { classes } = props;

  function isTimeSeries(facet) {
    return facet.time_names.length > 0;
  }

  function numCols(facet) {
    if (isTimeSeries(facet)) {
      if (facet.time_names.length <= 3) {
        return 1;
      } else if (facet.time_names.length <= 6) {
        return 2;
      } else {
        return 3;
      }
    } else {
      return 1;
    }
  }

  function getFacetDefinition(facet) {
    if (isTimeSeries(facet)) {
      return (
        <TimeSeriesFacet
          facet={facet}
          updateFacets={props.updateFacets}
          handleRemoveFacet={props.handleRemoveFacet}
          isExtraFacet={props.extraFacetEsFieldNames.includes(
            facet.es_field_name
          )}
          selectedFacetValues={props.selectedFacetValues}
          timeSeriesUnit={props.timeSeriesUnit}
        />
      );
    } else {
      return (
        <HistogramFacet
          facet={facet}
          updateFacets={props.updateFacets}
          handleRemoveFacet={props.handleRemoveFacet}
          isExtraFacet={props.extraFacetEsFieldNames.includes(
            facet.es_field_name
          )}
          selectedValues={props.selectedFacetValues.get(facet.es_field_name)}
        />
      );
    }
  }

  const facetsList = props.facets.map(facet => {
    return (
      // Can't set padding the normal way because it's overridden by
      // GridListTile's built-in "style=padding:2".
      <GridListTile
        classes={{ tile: classes.tile }}
        key={facet.es_field_name}
        style={{ padding: 0, marginBottom: 28 }}
        cols={numCols(facet)}
      >
        {getFacetDefinition(facet)}
      </GridListTile>
    );
  });
  return (
    // Can't set margin the normal way because it's overridden by
    // GridList's built-in "style=margin:-2px".
    <GridList
      classes={{ root: classes.root }}
      cols={3}
      cellHeight="auto"
      style={{ margin: "23px -10px -5px 15px" }}
    >
      {facetsList}
    </GridList>
  );
}

export default withStyles(styles)(FacetsGrid);
