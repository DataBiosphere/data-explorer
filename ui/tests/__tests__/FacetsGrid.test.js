import React from "react";
import FacetsGrid from "../../src/components/facets/FacetsGrid";
import { getFacetsList } from "../mockutils";

test("Renders correctly", () => {
  const wrapper = shallow(
    <FacetsGrid updateFacets={() => {}} facets={getFacetsList(2, 2)} />
  );
  expect(wrapper).toMatchSnapshot();
});
