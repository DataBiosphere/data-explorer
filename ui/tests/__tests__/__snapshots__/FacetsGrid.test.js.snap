// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<GridList
  cellHeight="auto"
  className="gridList"
  cols={3}
  padding={1}
>
  <GridTile
    actionPosition="right"
    cols={1}
    containerElement="div"
    key="Facet 1"
    rows={1}
    titleBackground="rgba(0, 0, 0, 0.4)"
    titlePosition="bottom"
  >
    <FacetCard
      facet={
        Facet {
          "name": "Facet 1",
          "values": Array [
            FacetValue {
              "count": 10,
              "name": "FacetValue 1",
            },
            FacetValue {
              "count": 20,
              "name": "FacetValue 2",
            },
          ],
        }
      }
      updateFacets={[Function]}
    />
  </GridTile>
  <GridTile
    actionPosition="right"
    cols={1}
    containerElement="div"
    key="Facet 2"
    rows={1}
    titleBackground="rgba(0, 0, 0, 0.4)"
    titlePosition="bottom"
  >
    <FacetCard
      facet={
        Facet {
          "name": "Facet 2",
          "values": Array [
            FacetValue {
              "count": 10,
              "name": "FacetValue 1",
            },
            FacetValue {
              "count": 20,
              "name": "FacetValue 2",
            },
          ],
        }
      }
      updateFacets={[Function]}
    />
  </GridTile>
</GridList>
`;
